### PR TITLE
fix build on Windows

### PR DESCRIPTION
**Pull request by Byron**
As per https://github.com/gitbutlerapp/gitbutler/pull/3541#issuecomment-2063644317, the windows build is broken.

This PR should fix it.

### Tasks

* [x] fix it
* [x] validate fix with quick CI check on Windows

### CODE REVIEW

File: .github/workflows/push.yaml [hunk 1]
```diff
@@ -211,3 +211,10 @@ jobs:
           allowed-skips: ${{ toJSON(needs) }}
           jobs: ${{ toJSON(needs) }}
 
+  check-rust-windows:
+    runs-on: windows-latest
```
Comment (Qix-) [inline .github/workflows/push.yaml:217]: can you add an `if` here:

```
    if: ${{ needs.changes.outputs.rust == 'true' }}
```

Comment (Qix-) [reply .github/workflows/push.yaml:217]: Just makes sure that our typescript-only changes don't take time to build all of rust on windows

Comment (Byron) [reply .github/workflows/push.yaml:217]: Makes sense, great catch! It's in there now.

**Comment by Qix-** [comment]
Ach sorry, one more thing: The windows check needs to be added to the `needs` list above in the final rust check job, otherwise failures won't fail the entire CI >.>

**Comment by Qix-** [comment]
Thank you 🙏🏻

**Pull request merged**